### PR TITLE
zapzap: init at 4.5.5.2

### PR DESCRIPTION
**Pull request by hatch01**
## Description of changes

- add the package zapzap
- add myself as a maintainer

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/za/zapzap/package.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  src = fetchFromGitHub {
+    owner = "zapzap-linux";
+    repo = "zapzap";
+    rev = version;
+    hash = "sha256-8IeFGTI+5kbeFGqH5DpHCY8pqzGhE48hPCEIKIe7jAM=";
+  };
+
+  propagatedBuildInputs = with python3Packages; [
+    setuptools
+    dbus-python
```
[the remark this answers is not in the document]
Comment (hatch01) [reply pkgs/by-name/za/zapzap/package.nix:23]: I tried to follow your suggestion, but I'm not a hundred percent sure that I understood.

**Comment by hatch01** [comment]
Thanks a lot for your reviews @SuperSandro2000 it's really pleasant to learn nixpkgs with such reviews.

**Comment by Xbz-24** [comment]
daily@nixos ~> zapzap
QPixmap::scaled: Pixmap is a null pixmap
QSystemTrayIcon::setVisible: No Icon set
Traceback (most recent call last):
  File "/nix/store/7q95dlv8fhhdzgny7qll02g3m97g66ca-python3.10-dbus-python-1.2.18/lib/python3.10/site-packages/dbus/bus.py", line 177, in activate_name_owner
    return self.get_name_owner(bus_name)
  File "/nix/store/7q95dlv8fhhdzgny7qll02g3m97g66ca-python3.10-dbus-python-1.2.18/lib/python3.10/site-packages/dbus/bus.py", line 361, in get_name_owner
    return self.call_blocking(BUS_DAEMON_NAME, BUS_DAEMON_PATH,
  File "/nix/store/7q95dlv8fhhdzgny7qll02g3m97g66ca-python3.10-dbus-python-1.2.18/lib/python3.10/site-packages/dbus/connection.py", line 652, in call_blocking
    reply_message = self.send_message_with_reply_and_block(
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.NameHasNoOwner: Could not get owner of name 'org.freedesktop.Notifications': no such name

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/nix/store/jwl0zvzhwkximmgm536h3pk1lsy60dyf-zapzap-4.5.5.2/bin/.zapzap-wrapped", line 9, in <module>
    sys.exit(main())
  File "/nix/store/jwl0zvzhwkximmgm536h3pk1lsy60dyf-zapzap-4.5.5.2/lib/python3.10/site-packages/zapzap/__main__.py", line 88, in main
    window = MainWindow(app)
  File "/nix/store/jwl0zvzhwkximmgm536h3pk1lsy60dyf-zapzap-4.5.5.2/lib/python3.10/site-packages/zapzap/controllers/main_window.py", line 40, in __init__
    self.zapHome = Home()
  File "/nix/store/jwl0zvzhwkximmgm536h3pk1lsy60dyf-zapzap-4.5.5.2/lib/python3.10/site-packages/zapzap/controllers/home.py", line 22, in __init__
    self.loadUsers()
  File "/nix/store/jwl0zvzhwkximmgm536h3pk1lsy60dyf-zapzap-4.5.5.2/lib/python3.10/site-packages/zapzap/controllers/home.py", line 30, in loadUsers
    button = UserContainer(self, user)
  File "/nix/store/jwl0zvzhwkximmgm536h3pk1lsy60dyf-zapzap-4.5.5.2/lib/python3.10/site-packages/zapzap/controllers/user_container.py", line 45, in __init__
    self.browser = Browser(user.id, self)
  File "/nix/store/jwl0zvzhwkximmgm536h3pk1lsy60dyf-zapzap-4.5.5.2/lib/python3.10/site-packages/zapzap/engine/browser.py", line 23, in __init__
    dbus.init(__appname__)
  File "/nix/store/jwl0zvzhwkximmgm536h3pk1lsy60dyf-zapzap-4.5.5.2/lib/python3.10/site-packages/zapzap/services/dbus_notify.py", line 55, in init
    proxy = bus.get_object(name, path)
  File "/nix/store/7q95dlv8fhhdzgny7qll02g3m97g66ca-python3.10-dbus-python-1.2.18/lib/python3.10/site-packages/dbus/bus.py", line 241, in get_object
    return self.ProxyObjectClass(self, bus_name, object_path,
  File "/nix/store/7q95dlv8fhhdzgny7qll02g3m97g66ca-python3.10-dbus-python-1.2.18/lib/python3.10/site-packages/dbus/proxies.py", line 250, in __init__
    self._named_service = conn.activate_name_owner(bus_name)
  File "/nix/store/7q95dlv8fhhdzgny7qll02g3m97g66ca-python3.10-dbus-python-1.2.18/lib/python3.10/site-packages/dbus/bus.py", line 182, in activate_name_owner
    self.start_service_by_name(bus_name)
  File "/nix/store/7q95dlv8fhhdzgny7qll02g3m97g66ca-python3.10-dbus-python-1.2.18/lib/python3.10/site-packages/dbus/bus.py", line 277, in start_service_by_name
    return (True, self.call_blocking(BUS_DAEMON_NAME, BUS_DAEMON_PATH,
  File "/nix/store/7q95dlv8fhhdzgny7qll02g3m97g66ca-python3.10-dbus-python-1.2.18/lib/python3.10/site-packages/dbus/connection.py", line 652, in call_blocking
    reply_message = self.send_message_with_reply_and_block(
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.Notifications was not provided by any .service files

I get this error when trying to run your script

**Comment by Xbz-24** [comment]
nixos configurations builds fine but when executing the binary I get the above error

File: pkgs/by-name/za/zapzap/package.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  meta = with lib; {
+    description = "WhatsApp desktop application for Linux ";
+    homepage = "https://zapzap-linux.github.io/";
+    mainProgram = "zapzap";
+    license = licenses.gpl3Only;
+    changelog = "https://github.com/zapzap-linux/zapzap/releases/tag/${version}";
+    maintainers = [ maintainers.eymeric ];
+  };
+}
+
```
Comment (SuperSandro2000) [inline pkgs/by-name/za/zapzap/package.nix:45]: there is an extra new line here

File: pkgs/by-name/za/zapzap/package.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    owner = "zapzap-linux";
+    repo = "zapzap";
+    rev = version;
+    hash = "sha256-8IeFGTI+5kbeFGqH5DpHCY8pqzGhE48hPCEIKIe7jAM=";
+  };
+
+  nativeBuildInputs = with python3Packages; [
+    setuptools
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/by-name/za/zapzap/package.nix:21]:
```suggestion

  buildInputs = [
    qt6.qtwayland
    qt6.qtsvg
  ];

  preBuild = ''
    export HOME=$(mktemp -d)
  '';

```

qtsvg is required to properly display the icons

File: pkgs/by-name/za/zapzap/package.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  src = fetchFromGitHub {
+    owner = "zapzap-linux";
+    repo = "zapzap";
+    rev = version;
+    hash = "sha256-8IeFGTI+5kbeFGqH5DpHCY8pqzGhE48hPCEIKIe7jAM=";
+  };
+
+  nativeBuildInputs = with python3Packages; [
+    setuptools
```
Comment (SuperSandro2000) [inline pkgs/by-name/za/zapzap/package.nix:19]:
```suggestion
    setuptools
    qt6.wrapQtAppsHook
```
this and the others fix startup on wayland

File: pkgs/by-name/za/zapzap/package.nix [hunk 5]
```diff
@@ -0,0 +1,45 @@
+{ lib
+, fetchFromGitHub
+, python3Packages
```
Comment (SuperSandro2000) [inline pkgs/by-name/za/zapzap/package.nix:3]:
```suggestion
, python3Packages
, qt6
```

File: pkgs/by-name/za/zapzap/package.nix [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    pyqt6-sip
+  ];
+
+  prePatch = "export HOME=$(mktemp -d)";
+
+  postInstall = ''
+    install -Dm555 share/applications/com.rtosta.zapzap.desktop -t $out/share/applications/
+    install -Dm555 share/icons/com.rtosta.zapzap.svg -t $out/share/icons/hicolor/scalable/
+  '';
+
```
Comment (SuperSandro2000) [inline pkgs/by-name/za/zapzap/package.nix:35]:
```suggestion

  # has no tests
  doCheck = false;

```

```
zapzap> Ran 0 tests in 0.000s
zapzap>
zapzap> OK
zapzap> Finished executing setuptoolsCheckPhase
```

We don't need to try to run tests if there are none

File: pkgs/by-name/za/zapzap/package.nix [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  propagatedBuildInputs = with python3Packages; [
+    dbus-python
+    pyqt6
+    pyqt6-webengine
+    pyqt6-sip
+  ];
+
+  prePatch = "export HOME=$(mktemp -d)";
+
```
Comment (SuperSandro2000) [inline pkgs/by-name/za/zapzap/package.nix:30]:
```suggestion
```

File: pkgs/by-name/za/zapzap/package.nix [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  version = "4.5.5.2";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "zapzap-linux";
+    repo = "zapzap";
+    rev = version;
+    hash = "sha256-8IeFGTI+5kbeFGqH5DpHCY8pqzGhE48hPCEIKIe7jAM=";
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/by-name/za/zapzap/package.nix:17]:
```suggestion

  patches = [
    # fixes that the tray icon was not installed
    (fetchpatch {
      url = "https://github.com/zapzap-linux/zapzap/pull/25/commits/4107b019555492e2c2692dd4c40553503047e6a8.patch";
      hash = "sha256-NQPGPXYFhVwsPXopEELG1n/f8yUj/74OFE1hTyt93Ng=";
    })
  ];

```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
I did some more testing and with all my suggestions wayland and icons should finally work.

File: pkgs/by-name/za/zapzap/package.nix (hunk 7 above)
Comment (hatch01) [reply pkgs/by-name/za/zapzap/package.nix:46]: I can not build without this.

Comment (SuperSandro2000) [reply pkgs/by-name/za/zapzap/package.nix:46]: I've moved out above to the correct phase because we don't need it that early and generally prePatch should only prepare files to apply patches.

File: pkgs/by-name/za/zapzap/package.nix [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+  '';
+
+  propagatedBuildInputs = with python3Packages; [
+    dbus-python
+    pyqt6
+    pyqt6-webengine
+    pyqt6-sip
+  ];
+
+
```
Comment (SuperSandro2000) [inline pkgs/by-name/za/zapzap/package.nix:49]:
```suggestion

```

**Review by SuperSandro2000 — APPROVED** [review]
Thanks for keeping up!

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by Xbz-24** [comment]
image

added the unstable package as overlay to my config 
why when I open zapzap I see this update chrome stuff? I dont even have chrome installed but last week was working fine, any quickfix?

**Comment by hatch01** [comment]
Realy strange I used it today and got no problems.
Zapzap use a webview so it is normal that it speek about chrome (chromium).
I will retest it

**Comment by SuperSandro2000** [comment]
> added the unstable package as overlay to my config

Do you just use the package or also all it's dependencies?

Unless whatsapp is doing A/B testing, then this does not happen on unstable.
